app/presenters/error_message/translator.rb

```diff
@@ -0,0 +1,84 @@
+#
+# This class is reponsible for retrieving plain
+# english error messages from a custom location,
+# typically error_messages.yml.
+#
+# It must retrieve english messages based on error keys
+# in various formats. These keys come from both custom validations,
+# rails validations, including nested attributes style validation
+# error messages.
+#
+
+module ErrorMessage
+  class Translator
+    include ErrorMessage::Helper
+
+    attr_reader :long_message,
+                :short_message,
+                :api_message,
+                :translations,
+                :translator
+
+    def initialize(translations, key, error)
+      @translations = translations
+      @key = Key.new(key)
+      @error = format_error(error)
+      @submodel_numbers = {}
+      @long_message = nil
+      @short_message = nil
+      @api_message = nil
+
+      translate!
+    end
+
+    private
+
+    def translate!
+      get_messages(@translations, @key, @error)
+
+      @long_message = substitute_submodel_numbers_and_names(@long_message)
+      @short_message = substitute_submodel_numbers_and_names(@short_message)
+      @api_message = substitute_submodel_numbers_and_names(@api_message)
+    end
+
+    def get_messages(translations, key, error)
+      if key.submodel?
+        translation_subset, submodel_key = extract_submodel_attribute(key)
+        get_messages(translation_subset, submodel_key, error)
+      elsif translation_exists?(translations, key, error)
+        @long_message = translations[key][error]['long']
+        @short_message = translations[key][error]['short']
+        @api_message = translations[key][error]['api']
```

Comment:
Use a factory to remove conditional:

@long_message, @short_mesasge, @api_message = GetMessage.new(@key, @error).messages
@long.messages ... = message_klass.new()

Reply:
agree to leave this conditional removal for now